doc/source/cluster/kubernetes/user-guides/upgrade-guide.md

```diff
@@ -26,54 +26,37 @@ If you want to understand the reasoning behind the CRD versioning plan, see [ray
-  * Example 1: Upgrade KubeRay from v0.6.0 to v1.0.0 without deleting the old CRD.
-    ```shell
-    # Install KubeRay v0.6.0 and CRD v1alpha1
-    helm install kuberay-operator kuberay/kuberay-operator --version 0.6.0
-
-    # The following instruction uninstalls only KubeRay v0.6.0. It does not uninstall CRD v1alpha1.
-    helm uninstall kuberay-operator
-
-    # Install KubeRay v1.0.0. Because the CRD already exists, the Helm instruction does not install the new CRD.
-    helm install kuberay-operator kuberay/kuberay-operator --version 1.0.0
-
-    # Check CRD
-    kubectl describe crd rayclusters.ray.io | grep v1
-    # You can only see "Name: v1alpha1", and cannot see "Name: v1".
-
-    # Install RayCluster v1.0.0 which uses CRD v1.
-    helm install raycluster kuberay/ray-cluster --version 1.0.0
-    # Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest:
-    # resource mapping not found for name: "raycluster-kuberay" namespace: "" from "": no
-    # matches for kind "RayCluster" in version "ray.io/v1"
-    # ensure CRDs are installed first
-    ```
-  * Example 2: Upgrade KubeRay from v0.6.0 to v1.0.0 with deleting the old CRD.
-    ```shell
-    # Install KubeRay v0.6.0 and CRD v1alpha1
-    helm install kuberay-operator kuberay/kuberay-operator --version 0.6.0
-
-    # The following instruction uninstalls only KubeRay v0.6.0. It does not uninstall CRD v1alpha1.
-    helm uninstall kuberay-operator
-
-    # Delete CRDs. Note that deleting the CRD causes a cascading deletion of custom resources.
-    kubectl delete crd rayclusters.ray.io
-    kubectl delete crd rayjobs.ray.io
-    kubectl delete crd rayservices.ray.io
-
-    # Install KubeRay v1.0.0 and new CRD including v1.
-    helm install kuberay-operator kuberay/kuberay-operator --version 1.0.0
-
-    # Check CRD
-    kubectl describe crd rayclusters.ray.io | grep v1
-    # You can see both "Name: v1alpha1" and "Name: v1".
-
-    # Install RayCluster v1.0.0 which uses CRD v1.
-    helm install raycluster kuberay/ray-cluster --version 1.0.0
-    ```
+Due to reliability and security implications of webhooks, KubeRay doesn't support a conversion webhook to convert v1alpha1 to v1 APIs.
+
+To upgrade the KubeRay version, follow these steps in order:
+1. Upgrade the CRD manifest, containing new fields added to the v1 CRDs.
+2. Upgrade the kuberay-operator image to the new version.
+3. Verify the success of the upgrade.
+
+The following is an example of upgrading KubeRay from v1.0.0 to v1.1.0:
+```
+# Upgrade the CRD to v1.1.0.
+# Note: This example uses kubectl because Helm doesn't support lifecycle management of CRDs.
+# See the Helm documentation for more details: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations
+$ kubectl replace -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.1.0"
```

Comment:
What's the reason for using `kubectl replace`? Is it to avoid the size limit of `kubectl.kubernetes.io/last-applied-configuration`?

Reply:
Yes. Using `kubectl apply --server-side` also works around the size limit, but I think `kubectl replace` here is fine because we do not expect multiple writers to the CRD anyways and the manifest in release should be the source of truth overwriting any previous updates by other writers

Reply:
With `kubectl replace` we need to be careful not to use the `--force` command because that will delete the CRD and recreate it. But I think it's fine here as long as we don't recommend it

Reply:
Using `kubectl apply --server-side --force-conflicts -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.0.0"` also works well, so I'm fine with putting either one in the docs

Reply:
Got it. Thank you for the explanation!